pymatgen_diffusion/neb/full_path_mapper.py

```diff
@@ -408,6 +415,41 @@ def get_path(self, max_val=100000):
                     raise RuntimeError("More than one edge matched in original graph.")
             yield u, path_hops
 
+    def add_properties_from_dict(self, unique_hop_dict):
+        """
+        A method to automatically populate the migration graph with properties data
+        from an external unique hop dictionary (e.g. if data was added to a migration
+        graph and stored as a unique hop dictionary for later use).
+
+        Args:
+            unique_hop_dict: unique hop dictionary containing data under the
+             properties field.
+        """
+        for uhop in unique_hop_dict.values():
+            uc_isite = PeriodicSite.from_dict(uhop["isite"])
+            uc_esite = PeriodicSite.from_dict(uhop["esite"])
+            m_path = MigrationPath(uc_isite, uc_esite, self.symm_structure)
+            data = uhop["properties"]
+
+            sites_eq = self.symm_structure.spacegroup.are_symmetrically_equivalent
+            for hop_label, hop in self.unique_hops.items():
+                if sites_eq([uc_isite], [hop["isite"]], symm_prec=self.symprec):
```

Comment:
so this is not actually enough to determine equality.
Please look at how the `__eq__` function in `MigrationHop` is implemented.
Also I'm not a big fan of adding this "properties key" the `data` dictionary of the edge is already a general-purpose.  So haveing another general-purpose key for the subdictionary is not really needed.  You can achieve what you need by comparing the "hop" attributes of the edges

Reply:
The MigrationHop.__eq__() function will be helpful. Now that has been added to pymatgen diffusion I can look into incorporating it.